cmake/SundialsSetupTPLs.cmake

```diff
@@ -78,6 +78,35 @@ if(ENABLE_HYPRE)
   list(APPEND SUNDIALS_TPL_LIST "HYPRE")
 endif()
 
+# ---------------------------------------------------------------
+# Find (and test) Kokkos
+# ---------------------------------------------------------------
+
+if(ENABLE_KOKKOS)
+  list(APPEND SUNDIALS_TPL_LIST "KOKKOS")
+  find_package(Kokkos REQUIRED
+               PATHS ${Kokkos_DIR})
+  set(KOKKOS_WORKS TRUE)
+  # We should be able to use Kokkos_DEVICES directly but it seems to get
+  # removed or unset in some CMake versions
+  set(KOKKOS_EXAMPLES_BACKENDS "${Kokkos_DEVICES}" CACHE STRING
+      "Kokkos backends to build examples with")
+  mark_as_advanced(FORCE KOKKOS_EXAMPLES_BACKENDS)
+  message(STATUS "Kokkos VERSION: ${Kokkos_VERSION}")
```

Comment:
It would be good to also display the `Kokkos_DEVICES` variable.

Reply:
This gets output by the Kokkos config file already. For some reason the variable kept getting truncated to just the last value when building the examples (at least in my testing) which is why I added the `KOKKOS_EXAMPLES_BACKENDS` option.